sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -59,6 +61,44 @@ public ClientLogger(String className) {
         logger = initLogger instanceof NOPLogger ? new DefaultLogger(className) : initLogger;
     }
 
+    /**
+     * Logs a formattable message that uses {@code {}} as the placeholder at {@code verbose} log level.
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * <p>Logging a message at verbose log level.</p>
+     *
+     *
+     * {@codesnippet com.azure.core.util.logging.clientlogger.log}
+     * @param logLevel Logging level for the log message.
+     * @param message The formattable message to log.
+     */
+    public void log(LogLevel logLevel, Supplier<String> message) {
+        if (message != null) {
+            performLogging(logLevel, false, DEFAULT_MESSAGE_FORMAT, message.get());
```

Comment:
`message.get()` will call the supplier to get the message immediately and will not defer the computation of the message until we determine if logging is enabled. The supplier should be called only after we check that logging is enabled at `logLevel`.

Reply:
I have implemented a solution which determines whether supplier evaluation is needed or not. I am not great fan of instance of checks, so the other solution would be to add one more argument to performLogging method.

Reply:
@srnagar @alzimmermsft  I have implemented a new solution for this issue, please check it. Also if possible can I make a  refactor in this class within this story? The performLogging is getting too complex.